record-builder-processor/src/main/java/io/soabase/recordbuilder/processor/InternalRecordBuilderProcessor.java

```diff
@@ -993,12 +1017,26 @@ public MyRecordBuilder p(T p) {
 
         var parameterSpecBuilder = ParameterSpec.builder(type.valueType(), component.name());
         methodSpec.addJavadoc("Set a new value for the {@code $L} record component in the builder\n", component.name())
-                .addStatement("this.$L = $T.of($L)", component.name(), type.typeName(), component.name());
+                .addStatement(getOptionalStatement(type), component.name(), type.typeName(), component.name());
         addConstructorAnnotations(component, parameterSpecBuilder);
+        filterOutNotNullAnnotationOnParameter(parameterSpecBuilder);
         methodSpec.addStatement("return this").addParameter(parameterSpecBuilder.build());
         builder.addMethod(methodSpec.build());
     }
 
+    private void filterOutNotNullAnnotationOnParameter(ParameterSpec.Builder parameterSpecBuilder) {
```

Comment:
I feel that this portion should be in a different PR. Let's focus this PR solely on the `ofNullable`. wdyt?

Reply:
I agree since the issue discussion has gone off on a tangent.
But the issue this tries to solve is when there's a `@NotNull Optional` field in record.
In that case current code without the method above would generate 

```java
setX(@NotNull X x) {
    Optional.ofNullable(x);
}
``` 

This isn't a problem in my company code base because we never validate builders but it will surprise some.
I've reverted the commit that adds this filter.

Reply:
I want to handle the null issue finally and completely which I hope to do with #111 - people seem to have strong opinions on this :D